.gitlab-ci.yml

```diff
@@ -0,0 +1,94 @@
+image: docker:latest
+services:
+  - docker:dind
+
+before_script:
+  - export KUBECONFIG=./Kubernetes-cluster.yaml
+
+variables:
+  DOCKER_DRIVER: overlay
+  DOCKER_TLS_CERTDIR: ""
+
+stages:
+  - package
```

Comment:
The package and deploy stages must be executed only for tag commits from master branch. The version number for the image is derived from the commit tags using ($CI_COMMIT_TAG). Please check webin-auth for reference.

Reply:
@Rajkumar-D I am not clear about this comment. Could you please let me know a bit more?

webin-auth and webin-portal handling of build and deployment is a bit different. The reason being, webin-auth has only one build and webin-portal has 3, each for each environment. webin-portal is no longer a spring boot application and we can utilize application-*.properties.

If you see the master branch build and deployment stages

only:
    refs:
      - master
      - tags
    variables:
      - $CI_COMMIT_BRANCH == "master"

Which is tag created on master only.

and if you see stages for dev and test you will the regex for dev and test tags because there are different tags for different environments and need to handled separately and its not same as the auth project.

Happy to discuss further in case you want to have a chat about it.

Reply:
@Rajkumar-D Can you merge the PR? I can see conflicts developing with more changes getting merged to master.